src/visitor.cpp

```diff
@@ -104,3 +121,65 @@ bool detail::visit(const cpp_entity& e, detail::visitor_callback_t cb, void* fun
     DEBUG_UNREACHABLE(detail::assert_handler{});
     return true;
 }
+
+bool detail::visit(const cpp_entity& e, visitor_filter_t fcb, void* filt_functor,
```

Comment:
So I took a crack at this, but I don't think this is possible the way you suggest due to the reasons outllined in our earlier discussion: https://github.com/foonathan/cppast/pull/18#commitcomment-21894751